tests/components/starlink/test_diagnostics.py

```diff
@@ -0,0 +1,29 @@
+"""Tests for Starlink diagnostics."""
+import aiohttp
+
+from homeassistant.components.starlink.const import DOMAIN
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+
+from .patchers import COORDINATOR_SUCCESS_PATCHER
+
+from tests.common import MockConfigEntry
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+
+
+async def test_diagnostics(hass: HomeAssistant, hass_client: aiohttp.client):
+    """Test generating diagnostics for a config entry."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_IP_ADDRESS: "1.2.3.4:0000"},
+    )
+
+    with COORDINATOR_SUCCESS_PATCHER:
+        entry.add_to_hass(hass)
+
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        diag = await get_diagnostics_for_config_entry(hass, hass_client, entry)
+
+        assert diag["status"]["id"] == "**REDACTED**"
```

Comment:
It is preferable to assert the whole dict.
If the dict is too big, you can create a fixture json file and then compare:
`assert diag = expected_diag`

Reply:
Makes sense, are these guidelines documented anywhere? I don't want to miss anything else

Reply:
It's part of the same guideline... we want to try to ensure that if a follow-up PR triggers a diag adjustment, then the user has to adjust the test and we can check that we are not now exposing sensitive information.

Usually, we also pass information from the config entry:
https://github.com/home-assistant/core/blob/e084fe490386d57c8a85717d00a63d47cba4ca39/tests/components/accuweather/test_diagnostics.py#L21-L27